sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -288,10 +293,12 @@ private[hive] class SparkExecuteStatementOperation(
       }
 
       sqlContext.sparkContext.setJobGroup(statementId, substitutorStatement, forceCancel)
-      result = sqlContext.sql(statement)
-      logDebug(result.queryExecution.toString())
+      val df = sqlContext.sql(statement)
+      logDebug(df.queryExecution.toString())
       HiveThriftServer2.eventManager.onStatementParsed(statementId,
-        result.queryExecution.toString())
+        df.queryExecution.toString())
+      rdd = df.rdd
```

Comment:
`df.rdd` has tripped me in the past - the fact that it actually ends up executing the prefix dag was surprising to me !

Reply:
the changes here cannot trigger the eager clean for all cases, for example, CTAS, where the `S` contains a shuffle, will not be able to be cleared. The possible and proper place may be SparkPlan